src/rules/noVarKeywordRule.ts

```diff
@@ -37,46 +38,50 @@ export class Rule extends Lint.Rules.AbstractRule {
     public static FAILURE_STRING = "Forbidden 'var' keyword, use 'let' or 'const' instead";
 
     public apply(sourceFile: ts.SourceFile): Lint.RuleFailure[] {
-        const noVarKeywordWalker = new NoVarKeywordWalker(sourceFile, this.getOptions());
-        return this.applyWithWalker(noVarKeywordWalker);
+        return this.applyWithFunction(sourceFile, walk);
     }
 }
 
-class NoVarKeywordWalker extends Lint.RuleWalker {
-    public visitVariableStatement(node: ts.VariableStatement) {
-        if (!Lint.hasModifier(node.modifiers, ts.SyntaxKind.DeclareKeyword)
-                && !Lint.isBlockScopedVariable(node)) {
-            this.reportFailure(node.declarationList);
-        }
-
-        super.visitVariableStatement(node);
-    }
+function walk(ctx: Lint.WalkContext<void>): void {
+    const { sourceFile } = ctx;
+    return ts.forEachChild(ctx.sourceFile, function cb(node: ts.Node): void {
+        switch (node.kind) {
```

Comment:
you could simplify the condition and replace the whole switch statement with this:
```ts
if (isVariableDeclarationList(node) && !isBlockScopedVariableDeclarationList(node) &&
    (node.parent!.kind !== ts.SyntaxKind.VariableStatement || !isGlobalVarDeclaration(node.parent!)))
  fail(node);
```

you could even consider inlining `fail`